cli/src/main/java/org/commonjava/maven/ext/manip/Cli.java

```diff
@@ -0,0 +1,188 @@
+package org.commonjava.maven.ext.manip;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.execution.DefaultMavenExecutionRequest;
+import org.apache.maven.execution.DefaultMavenExecutionResult;
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.mae.MAEException;
+import org.apache.maven.mae.app.AbstractMAEApplication;
+import org.codehaus.plexus.PlexusContainer;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.commonjava.maven.ext.manip.io.PomIO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Properties;
+
+@Component( role = Cli.class )
+public class Cli extends AbstractMAEApplication
+{
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Requirement
+    private PlexusContainer container;
+
+    @Requirement
+    private ManipulationSession session;
+
+    @Requirement
+    private ManipulationManager manipulationManager;
+
+    @Requirement
+    private PomIO pomIO;
+
+    private File target = new File( System.getProperty( "user.dir" ), "pom.xml" );
+
+    private Properties userProps;
+
+    public static void main( String[] args )
+    {
+        Cli pmeCli = new Cli();
+        try
+        {
+            pmeCli.load();
+        }
+        catch ( MAEException e )
+        {
+            pmeCli.logger.debug( "Caught problem instantiating ", e );
+            System.err.println( "Unable to start Cli subsystem" );
+            System.exit( 1 );
+        }
+
+        pmeCli.run( args );
+    }
+
+    private void run( String[] args )
+    {
+        Options options = new Options();
+        options.addOption( "h", false, "Print this help message." );
+        options.addOption( Option.builder( "f" )
+                                 .longOpt( "file" )
+                                 .hasArgs()
+                                 .numberOfArgs( 1 )
+                                 .desc( "POM file" )
+                                 .build() );
+        options.addOption( Option.builder( "D" )
+                                 .hasArgs()
+                                 .numberOfArgs( 2 )
+                                 .valueSeparator( '=' )
+                                 .desc( "Java Properties" )
+                                 .build() );
+
```

Comment:
Using commons-cli rather than args4j as that allows e.g. 
`java -jar <jar> -Doption=option -Doption2=option2`
rather than args4j which I could only get `-D option=option -Doption2=option`